app/code/Vendor/Voucher/Model/VoucherManagement.php

```diff
@@ -0,0 +1,163 @@
+    /**
+     * @var VoucherResourceFactory
+     */
+    private $voucherResourceFactory;
+    /**
+     * @var VoucherCollectionFactory
+     */
+    private $voucherCollectionFactory;
+    /**
+     * @var CustomerModelFactory
+     */
+    private $customerModelFactory;
+
+    /**
+     * VoucherManagement constructor.
+     * @param VoucherStatusCollectionFactory $voucherStatusCollectionFactory
+     * @param VoucherStatusFactory $voucherStatusModelFactory
+     * @param VoucherStatusResourceModel $voucherStatusResourceFactory
+     * @param VoucherFactory $voucherModelFactory
+     * @param VoucherResourceFactory $voucherResourceFactory
+     * @param VoucherCollectionFactory $voucherCollectionFactory
+     * @param CustomerModelFactory $customerModelFactory
+     */
+    public function __construct(
+        VoucherStatusCollectionFactory $voucherStatusCollectionFactory,
+        VoucherStatusModelFactory $voucherStatusModelFactory,
+        VoucherStatusResourceModel $voucherStatusResourceFactory,
+        VoucherModelFactory $voucherModelFactory,
+        VoucherResourceFactory $voucherResourceFactory,
+        VoucherCollectionFactory $voucherCollectionFactory,
+        CustomerModelFactory $customerModelFactory
+    ) {
+        $this->voucherStatusCollectionFactory = $voucherStatusCollectionFactory;
+        $this->voucherStatusModelFactory = $voucherStatusModelFactory;
+        $this->voucherStatusResourceFactory = $voucherStatusResourceFactory;
+        $this->voucherModelFactory = $voucherModelFactory;
+        $this->voucherResourceFactory = $voucherResourceFactory;
+        $this->voucherCollectionFactory = $voucherCollectionFactory;
+        $this->customerModelFactory = $customerModelFactory;
+    }
+
+    public function createVoucherStatus()
+    {
+        $content = trim(file_get_contents("php://input"));
+        $status = json_decode($content, true)['status'];
+        $voucherStatus = $this->voucherStatusModelFactory->create();
+        $voucherStatus->setData(['status_code' => $status]);
+        $voucherStatus->save();
+        return $voucherStatus->getId();
+    }
+
+    public function deleteVoucherStatus($id)
+    {
+        $voucherStatus = $this->voucherStatusModelFactory->create();
+        try {
+            $voucherStatus->setId($id)->delete();
+        } catch (\Exception $e) {
+            throw new \Exception($e->getMessage());
+        }
+
+
+        return true;
+    }
+
+    public function getAllVoucherStatuses()
+    {
+        $data = [];
+        $voucherStatusCollection = $this->voucherStatusCollectionFactory->create();
+        /**
+         * @var \Vendor\Voucher\Model\VoucherStatus $voucherStatus
+         */
+        foreach ($voucherStatusCollection as $voucherStatus) {
+            $data[] = $voucherStatus->getStatusCode();
+        }
+        return $data;
+    }
+
+    public function createVoucher()
+    {
+        $content = trim(file_get_contents("php://input"));
```

Comment:
"php://input" - it's a reading data direct from the stream.
You should never use it directly due to security. Because you don't know, what exactly it contains.
Params are already passed to the method via arguments.